hooks/useMapLayer.ts

```diff
@@ -11,22 +11,28 @@ import {ZodError} from 'zod';
 
 export const useMapLayer = (center_id: string) => {
   const clientProps = React.useContext<ClientProps>(ClientContext);
-  return useQuery<MapLayer, AxiosError | ZodError>(['map-layer', center_id], async () => {
-    const url = `${clientProps.nationalAvalancheCenterHost}/v2/public/products/map-layer/${center_id}`;
-    const {data} = await axios.get(url);
+  return useQuery<MapLayer, AxiosError | ZodError>(
+    ['map-layer', center_id],
+    async () => {
+      const url = `${clientProps.nationalAvalancheCenterHost}/v2/public/products/map-layer/${center_id}`;
+      const {data} = await axios.get(url);
 
-    const parseResult = mapLayerSchema.safeParse(data);
-    if (parseResult.success === false) {
-      console.warn('unparsable map layer', url, parseResult.error, JSON.stringify(data, null, 2));
-      Sentry.Native.captureException(parseResult.error, {
-        tags: {
-          zod_error: true,
-          center_id,
-        },
-      });
-      throw parseResult.error;
-    } else {
-      return parseResult.data;
-    }
-  });
+      const parseResult = mapLayerSchema.safeParse(data);
+      if (parseResult.success === false) {
+        console.warn('unparsable map layer', url, parseResult.error, JSON.stringify(data, null, 2));
+        Sentry.Native.captureException(parseResult.error, {
+          tags: {
+            zod_error: true,
+            center_id,
+          },
+        });
+        throw parseResult.error;
+      } else {
+        return parseResult.data;
+      }
+    },
+    {
+      enabled: !!center_id,
```

Comment:
seems like we should have a predefined data type for `Center`, as opposed to potentially letting it be any string

Reply:
like, I'm not sure how we would ever let an empty string in here, but we could enforce that with TS

Reply:
Done in [bc598a9](https://github.com/stevekuznetsov/avalanche-forecast/pull/51/commits/bc598a95ffee9590d3d7dc2e83a849fcba9998a4)